### PR TITLE
Feat: Proxied wallet details

### PR DESCRIPTION
**Pull request by egor0798**
Changes:
- Added wallet details for delegated to you (proxied) wallet
- Updated `WalletCardLg` height

***

image

### CODE REVIEW

File: src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  className?: string;
+};
+
+export const ProxiesList = ({ walletId, className, canCreateProxy = true }: Props) => {
+  const { t } = useI18n();
+
+  const proxyAccounts = useUnit(walletProviderModel.$proxyAccounts);
+  const proxyChainGroups = useUnit(proxyModel.$proxyChainGroupStore)[walletId];
+
+  if (!proxyAccounts || !proxyAccounts.length) return <EmptyProxyList className={className} />;
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx:28]: `ProxiesList` responsibility is to display the list of proxies
It must know nothing about `EmptyList`
The condition that controls what component to display must be before `ProxiesList` or `EmptyProxiesList`

File: src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    return (
+      <DropdownIconButton name="more" className="ml-2">
+        <DropdownIconButton.Items>
+          <DropdownIconButton.Item>
+            <ExplorersPopover
+              address={proxyAccount.accountId}
+              explorers={chain.explorers || []}
+              addressPrefix={chain.addressPrefix}
+              className="-mt-10 -mr-1"
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx:54]: I don't like that for strange situations with negative margins we are forced to add `className` though we easily lived without it for a month or so
Could you provide a screenshot so we could discuss details and come up with another solution?

File: src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  const getProxyActionButton = (proxyAccount: ProxyAccount, chain: Chain) => {
+    const proxiedAddress = toAddress(proxyAccount.proxiedAccountId, { prefix: chain.addressPrefix });
+
+    return (
+      <DropdownIconButton name="more" className="ml-2">
+        <DropdownIconButton.Items>
+          <DropdownIconButton.Item>
+            <ExplorersPopover
+              address={proxyAccount.accountId}
+              explorers={chain.explorers || []}
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx:52]: It's `explorers = []` by default or it doesn't work without `|| []`?

File: src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    <div className={cnTw('flex flex-col', className)}>
+      <div className="flex items-center px-5 pb-2">
+        <FootnoteText className="flex-1 px-2 text-text-tertiary">{t('accountList.addressColumn')}</FootnoteText>
+      </div>
+
+      <ul className="flex flex-col px-5 divide-y divide-divider overflow-y-auto overflow-x-hidden">
+        {proxyChainGroups.map((chainGroup) => {
+          const { chainId, totalDeposit } = chainGroup;
+
+          const chain = chainsService.getChainById(chainId);
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx:90]: `getChainById` is not a good function, I'll deprecate it one day
Better to get `networkModel.chains` and `chains[chainId]`

File: src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+              <Accordion isDefaultOpen>
+                <Accordion.Button buttonClass="p-2 rounded hover:bg-action-background-hover focus:bg-action-background-hover">
+                  <div className="flex gap-x-2 items-center justify-between flex-1 pr-2">
+                    <ChainTitle className="flex-1" fontClass="text-text-primary" chain={chain} />
+                    <HelpText className="text-text-tertiary">
+                      {t('walletDetails.common.proxyDeposit')}
+                      &nbsp;
+                      <AssetBalance
+                        value={totalDeposit}
+                        asset={chain?.assets[0]}
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx:105]:
```suggestion
                        asset={chain.assets[0]}
```

File: src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+                        showIcon={false}
+                        className="text-help-text"
+                      />
+                    </HelpText>
+                  </div>
+                </Accordion.Button>
+                <Accordion.Content>
+                  <ul className="flex flex-col gap-y-2">
+                    {(proxiesByChain[chainId] || []).map((proxy) => (
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx:114]: At line `92` we've checked that `proxiesByChain[chainId]` has items

File: src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+                      <li className="px-2 py-1.5" key={proxy.accountId}>
+                        <ProxyAccountComponent
+                          accountId={proxy.accountId}
+                          proxyType={proxy.proxyType}
+                          addressPrefix={chain?.addressPrefix}
+                          actionButton={getProxyActionButton(proxy, chain)}
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx:120]: You can create local component `ProxyAccountWithActions` move there `ProxyAccount` filled by `DropdownButton` actions without an additional function, pure jsx

File: src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx [hunk 8]
```diff
@@ -25,7 +25,10 @@ import {
   Icon,
   SmallTitleText,
   StatusModal,
+  Tabs,
 } from '@shared/ui';
+import { TabItem } from '@shared/ui/Tabs/common/types';
+import { ProxiesList } from '@widgets/WalletDetails/ui/ProxiesList';
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx:31]:
```suggestion
import { ProxiesList } from './ProxiesList';
```

File: src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx [hunk 9]
```diff
@@ -115,6 +118,45 @@ export const WalletConnectDetails = ({ wallet, accounts, onClose }: Props) => {
     </DropdownIconButton>
   );
 
+  const tabItems: TabItem[] = [
+    {
+      id: 'accounts',
+      title: t('walletDetails.common.accountTabTitle'),
+      panel: (
+        <>
+          {wcDetailsUtils.isNotStarted(reconnectStep, wallet.isConnected) && (
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx:127]: Would be great to create a dedicated component

File: src/renderer/widgets/WalletDetails/ui/WalletDetailsProvider.tsx [hunk 10]
```diff
@@ -8,6 +8,8 @@ import { MultishardWalletDetails } from './MultishardWalletDetails';
 import { VaultWalletDetails } from './VaultWalletDetails';
 import { walletProviderModel } from '../model/wallet-provider-model';
 import { walletUtils } from '@entities/wallet';
+import { ProxiedWalletDetails } from '@widgets/WalletDetails/ui/ProxiedWalletDetails';
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/WalletDetailsProvider.tsx:11]:
```suggestion
import { ProxiedWalletDetails } from './ProxiedWalletDetails';
```

File: src/renderer/widgets/WalletDetails/ui/VaultWalletDetails.tsx [hunk 11]
```diff
@@ -1,21 +1,23 @@
...
 import { VaultMap } from '../lib/types';
 import { ShardsList } from './ShardsList';
 import { vaultDetailsModel } from '../model/vault-details-model';
 import { walletDetailsUtils } from '../lib/utils';
-import { KeyConstructor, ImportKeysModal, DerivationsAddressModal } from '@features/wallets';
+import { DerivationsAddressModal, ImportKeysModal, KeyConstructor } from '@features/wallets';
 import { RenameWalletModal } from '@features/wallets/RenameWallet';
 import { ForgetWalletModal } from '@features/wallets/ForgetWallet';
+import { TabItem } from '@shared/ui/Tabs/common/types';
+import { ProxiesList } from '@widgets/WalletDetails/ui/ProxiesList';
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/VaultWalletDetails.tsx:20]:
```suggestion
import { ProxiesList } from './ProxiesList';
```

**Review by tuul-wq — CHANGES_REQUESTED** [review]

File: src/renderer/shared/assets/images/arrows/curve-left-right.svg
[the remark this answers is not in the document]
Comment (egor0798) [reply src/renderer/shared/assets/images/arrows/curve-left-right.svg:1]: yes "arrow-curve-left-right"

File: src/renderer/shared/ui/Icon/data/functionals.tsx [hunk 12]
```diff
@@ -118,6 +119,7 @@ const FunctionalImages = {
   more: { img: MoreImg, svg: MoreSvg },
   rename: { img: RenameImg, svg: RenameSvg },
   forget: { img: ForgetImg, svg: ForgetSvg },
+  emptyListFile: { img: EmptyListImg, svg: EmptyListSvg },
```
[the remark this answers is not in the document]
Comment (egor0798) [reply src/renderer/shared/ui/Icon/data/functionals.tsx:122]: We already have "emptyList" icon, but while looking fo that I found that the same icon I need here is already added and called "document"

File: src/renderer/widgets/WalletDetails/ui/MultishardWalletDetails.tsx [hunk 13]
```diff
@@ -62,12 +63,16 @@ export const MultishardWalletDetails = ({ wallet, accounts, onClose }: Props) =>
...
+      panel: <MultishardAccountsList accounts={accounts} className="h-[409px]" />,
     },
     {
       id: 'proxies',
       title: t('walletDetails.common.proxiesTabTitle'),
-      panel: <ProxiesList walletId={wallet.id} className="h-[393px] mt-4" />,
+      panel: hasProxies ? (
+        <ProxiesList walletId={wallet.id} className="h-[393px] mt-4" />
+      ) : (
+        <EmptyProxyList className="h-[393px] mt-4" />
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/MultishardWalletDetails.tsx:74]: Let's name it `<NoProxiesAction />` and move to local `ui` folder

File: src/renderer/widgets/WalletDetails/ui/MultisigWalletDetails.tsx [hunk 14]
```diff
@@ -12,7 +13,9 @@ import { WalletFiatBalance } from '@features/wallets/WalletSelect/ui/WalletFiatB
 import { IconNames } from '@shared/ui/Icon/data';
 import { RenameWalletModal } from '@features/wallets/RenameWallet';
 import { ForgetWalletModal } from '@features/wallets/ForgetWallet';
-import { ProxiesList } from '@widgets/WalletDetails/ui/ProxiesList';
+import { ProxiesList } from './ProxiesList';
+import { walletProviderModel } from '@widgets/WalletDetails/model/wallet-provider-model';
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/MultisigWalletDetails.tsx:17]: import can be shortened

File: src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx [hunk 15]
```diff
@@ -25,8 +25,6 @@ export const ProxiesList = ({ walletId, className, canCreateProxy = true }: Prop
   const proxyAccounts = useUnit(walletProviderModel.$proxyAccounts);
   const proxyChainGroups = useUnit(proxyModel.$proxyChainGroupStore)[walletId];
 
-  if (!proxyAccounts || !proxyAccounts.length) return <EmptyProxyList className={className} />;
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/ProxiesList.tsx]: Excellent 👍

File: src/renderer/widgets/WalletDetails/ui/VaultWalletDetails.tsx [hunk 16]
```diff
@@ -18,6 +17,8 @@ import { RenameWalletModal } from '@features/wallets/RenameWallet';
 import { ForgetWalletModal } from '@features/wallets/ForgetWallet';
 import { TabItem } from '@shared/ui/Tabs/common/types';
 import { ProxiesList } from '@widgets/WalletDetails/ui/ProxiesList';
+import { walletProviderModel } from '@widgets/WalletDetails/model/wallet-provider-model';
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/VaultWalletDetails.tsx:20]: shorten import

**Review by tuul-wq — CHANGES_REQUESTED** [review]

File: src/renderer/widgets/WalletDetails/ui/wallets/MultishardWalletDetails.tsx [hunk 17]
```diff
@@ -5,13 +5,13 @@ import { useModalClose, useToggle } from '@shared/lib/hooks';
...
-import type { MultishardMap } from '../lib/types';
-import { walletDetailsUtils } from '../lib/utils';
+import type { MultishardMap } from '../../lib/types';
+import { walletDetailsUtils } from '../../lib/utils';
 import { RenameWalletModal } from '@features/wallets/RenameWallet';
 import { IconNames } from '@shared/ui/Icon/data';
 import { ForgetWalletModal } from '@features/wallets/ForgetWallet';
 import { TabItem } from '@shared/ui/Tabs/common/types';
-import { ProxiesList } from '@widgets/WalletDetails/ui/ProxiesList';
+import { ProxiesList } from '@widgets/WalletDetails/ui/components/ProxiesList';
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/wallets/MultishardWalletDetails.tsx:14]: Short imports

**Review by tuul-wq — CHANGES_REQUESTED** [review]

File: src/renderer/widgets/WalletDetails/ui/components/WalletConnectAccounts.tsx [hunk 18]
```diff
@@ -25,11 +25,11 @@ type Props = {
 export const WalletConnectAccounts = ({ wallet, accounts }: Props) => {
   const { t } = useI18n();
 
+  const chains = useUnit(networkModel.$chains);
   const reconnectStep = useUnit(wcDetailsModel.$reconnectStep);
 
-  // TODO: Rework with https://app.clickup.com/t/8692ykm3y
   const accountsList = useMemo(() => {
-    const sortedChains = chainsService.getChainsData({ sort: true });
+    const sortedChains = chainsService.sortChains(Object.values(chains));
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/components/WalletConnectAccounts.tsx:32]: Chains are already sorted in `networkModel`
line 12 `const chains = chainsService.getChainsMap({ sort: true });`

File: src/renderer/entities/wallet/ui/MultishardAccountsList/MultishardAccountsList.tsx [hunk 19]
```diff
@@ -1,20 +1,22 @@
...
 type Props = {
   accounts: Map<BaseAccount, Record<ChainId, ChainAccount[]>>;
-  chains: Chain[];
   className?: string;
 };
 
-export const MultishardAccountsList = ({ accounts, chains, className }: Props) => {
+export const MultishardAccountsList = ({ accounts, className }: Props) => {
   const { t } = useI18n();
+  const chains = useUnit(networkModel.$chains);
```
Comment (tuul-wq) [inline src/renderer/entities/wallet/ui/MultishardAccountsList/MultishardAccountsList.tsx:19]: Please move `chains` to props

File: src/renderer/shared/core/index.ts [hunk 20]
```diff
@@ -52,14 +52,15 @@ export type { Validator } from './types/validator';
 export { RewardsDestination } from './types/stake';
 export type { Stake, Unlocking } from './types/stake';
 
+export { ProxyType, ProxyVariant } from './types/proxy';
```
Comment (tuul-wq) [inline src/renderer/shared/core/index.ts:55]: This import was grouped with other Proxy types, now it's separated
Please move it back

File: src/renderer/widgets/WalletDetails/ui/components/NoProxiesAction.tsx [hunk 21]
```diff
@@ -0,0 +1,24 @@
+import { Icon, SmallTitleText, FootnoteText, Button } from '@shared/ui';
+import { useI18n } from '@app/providers';
+import { ProxyPopover } from '@entities/proxy/ui/ProxyPopover';
+import { cnTw } from '@shared/lib/utils';
+
+type Props = {
+  className?: string;
+};
+
+export const NoProxiesAction = ({ className }: Props) => {
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/components/NoProxiesAction.tsx:10]: Nice 👍

**Review by tuul-wq — COMMENTED** [review]
Requires one last change

**Review by tuul-wq — APPROVED** [review]

File: src/renderer/entities/proxy/lib/__tests__/utils.test.ts [hunk 22]
```diff
@@ -1,3 +1,5 @@
+import { expect } from '@playwright/test';
```
Comment (tuul-wq) [inline src/renderer/entities/proxy/lib/__tests__/utils.test.ts:1]: We use jest to `expect` not `playwright`

File: src/renderer/entities/proxy/lib/__tests__/utils.test.ts [hunk 23]
```diff
@@ -57,4 +59,41 @@ describe('entities/proxy/lib/utils', () => {
...
+        proxiedAccountId: '0x02',
+        chainId: '0x05',
+        proxyType: ProxyType.NON_TRANSFER,
+        delay: 0,
+      },
+    ];
+
+    const sortedAccounts = proxyUtils.sortAccountsByProxyType(proxyAccounts);
+
+    expect(sortedAccounts.map((a) => a.proxyType)).toEqual([
```
Comment (tuul-wq) [inline src/renderer/entities/proxy/lib/__tests__/utils.test.ts:93]: Why not this?
```javascript
expect(sortedAccounts).toEqual([proxyAccounts[2], proxyAccounts[0], proxyAccounts[1]]);
```

File: src/renderer/widgets/WalletDetails/model/__tests__/walet-provider-model.test.ts [hunk 24]
```diff
@@ -0,0 +1,82 @@
+import { allSettled, fork } from 'effector';
+import { expect } from '@playwright/test';
+
+import { walletModel } from '@entities/wallet';
+import { walletSelectModel } from '@features/wallets';
+import { walletProviderModel } from '../wallet-provider-model';
+import { storageService } from '@shared/api/storage';
+import { proxyModel } from '@entities/proxy';
+import { walletProviderMock } from '@widgets/WalletDetails/model/__tests__/wallet-provider.mock';
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/model/__tests__/walet-provider-model.test.ts:9]:
```suggestion
import { walletProviderMock } from './wallet-provider.mock';
```

**Review by tuul-wq — APPROVED** [review]
With tiny suggestion

**Pull request merged**